airbyte-integrations/bases/base-normalization/README.md

```diff
@@ -1,11 +1,147 @@
+  which removes the records which were synced multiple times.
+* If normalization fails partway through, such that (for example) the SCD model is updated but the final table is not, and then the sync
+  is retried, then the source will not re-emit any old records (because the destination will have emitted a state message ack-ing
+  all of the records). If the retry emits some new records, then normalization will append them to the SCD table as usual
+  (because, from the SCD's point of view, this is just a normal sync). Then the final table's latest `__airbyte_emitted_at`
+  will be older than the original attempt, so it will pull both the new records _and_ the first attempt's records from the SCD table.
+
+## Developer workflow
+
+At a high level, this is the recommended workflow for updating base-normalization:
+1. Manually edit the models in `integration_tests/normalization_test_output/postgres/test_simple_streams/models/generated`.
+   Run `dbt compile` and manually execute the SQL queries. This requires manual setup and validation, but allows you to quickly experiment
+   with different inputs.
+    1. You can substitute your preferred database/warehouse. This document will use Postgres because it's easy to set up.
+1. Once `dbt run` succeeds, edit `stream_processor.py` until it generates the models you hand-wrote in step 1.
+1. Run the `test_normalization[DestinationType.POSTGRES-test_simple_streams]` integration test case.
+1. Run the full [integration test suite](#integration-tests).
+1. Commit the changes in `integration_tests/normalization_test_output`.
+
+### Setting up your environment
+
+If you have a fullly-featured Python dev environment, you can just set a breakpoint at [this line]([integration_tests/test_normalization.py#L105](https://github.com/airbytehq/airbyte/blob/17ee3ad44ff71164765b97ff439c7ffd51bf9bfe/airbyte-integrations/bases/base-normalization/integration_tests/test_normalization.py#L108))
+and run the `test_normalization[DestinationType.POSTGRES-test_simple_streams]` test case. You can terminate the run after it hits the
+breakpoint. This will start Postgres in a Docker container with some prepopulated data and configure profiles.yml to match the container.
+
+Otherwise, you can run this command:
+```shell
+docker run \
+  --rm \
+  --name "normalization_dev_postgres" \
+  -e "integration-tests" \
+  -e "integration-tests" \
+  -p "9001:5432" \
+  -d \
+  marcosmarxm/postgres-ssl:dev \
+  -c ssl=on \
+  -c ssl_cert_file=/var/lib/postgresql/server.crt \
+  -c ssl_key_file=/var/lib/postgresql/server.key \
+```
+
+Then you'll need to edit `integration_tests/normalization_test_output/postgres/test_simple_streams/profiles.yml` and set the port to 9001.
+
+### Running dbt
+
+Once you have a database available, you can run dbt commands. We recommend running dbt from inside the `airbyte/normalization:latest` image.
+This saves you the effort of installing dbt and reconfiguring dbt_project.yml.
```

Comment:
Otherwise, you need to make sure to install the same dbt version as the one used in normalization (because the dbt_project.yml may be tied to a specific dbt version syntax)